NuRadioMC/EvtGen/generator.py

```diff
@@ -1060,6 +1041,11 @@ def get_inverse_spectrum(N, gamma):
 
     print("number of fiducial showers", len(data_sets_fiducial['flavors']))
 
+    if len(data_sets_fiducial['event_ids']) == 0:
+        for key, value in data_sets.items():
+            data_sets_fiducial[key] = np.array( [data_sets[key][0]] )
+        data_sets_fiducial['flavors'] = np.array( [14] )
```

Comment:
can you explain these lines? It seems as if you were adding one muon event in case there are no events in the fiducial volume. This doesn't seem right.

Reply:
The hdf5 writing function crashes when data_sets_fiducial is empty, and this make my scripts crash. For the muons, we do want files that are empty, they're just telling you that there are no events for that particular try. The easiest option was to add a muon neutrino event (that is not a muon, so it can be spotted easily), with zero shower energy (zero inelasticity), and at the top of the atmosphere so it wouldn't trigger anyway.

If you prefer a different solution just tell me.

Reply:
this seems pretty hacked. Although is should work fine... I don't know a better way at the moment. Do you need to write a file? Or would not writing a file at all be an option? Then I would just implement this saveguard in the writer function. 

Can you add least add a comment to the source code explaining what you do and why?

Reply:
Let's assume you don't write a file. Then you are biasing your effective areas because you're only taking files with shower-inducing events and ignoring the ones that don't, artificially increasing the effective area. However, it is not that critical, since at these energies the effective areas are usually zero.

I am going to add a comment right now.